functions/src/create-contract.ts

```diff
@@ -94,58 +71,47 @@ export const createContract = newEndpoint(['POST'], async (req, _res) => {
 
   const userContractsCreatedTodaySnapshot = await firestore
     .collection(`contracts`)
-    .where('creatorId', '==', creator.id)
+    .where('creatorId', '==', user.id)
     .where('createdTime', '>=', freeMarketResetTime)
     .get()
   console.log('free market reset time: ', freeMarketResetTime)
   const isFree = userContractsCreatedTodaySnapshot.size === 0
 
   const ante = FIXED_ANTE
 
-  if (
-    ante === undefined ||
-    ante < MINIMUM_ANTE ||
-    (ante > creator.balance && !isFree) ||
-    isNaN(ante) ||
-    !isFinite(ante)
-  )
-    throw new APIError(400, 'Invalid ante')
-
   console.log(
     'creating contract for',
-    creator.username,
+    user.username,
     'on',
     question,
     'ante:',
     ante || 0
   )
 
   const slug = await getSlug(question)
-
   const contractRef = firestore.collection('contracts').doc()
-
   const contract = getNewContract(
     contractRef.id,
     slug,
-    creator,
+    user,
     question,
     outcomeType,
     description,
-    initialProb,
+    initialProb ?? 0,
```

Comment:
Shouldn't we not set initialProb for non-binary markets?

Reply:
So what happened here is:

- The old code was typing `initialProb` as `any`, even though it could be `undefined`.
- So it passed `undefined` for non-binary markets into `getNewContract`, even though `getNewContract` claims to only take a `number`.
- However, inside `getNewContract`, it only uses `initialProb` or sets it on the contract in binary markets.

So I think this code actually behaves identically in the end to the old code. Of course, it would be better if `getNewContract` was better designed so it could accept `undefined` in cases where it was a binary contract. This should be doable with some type refactoring. But in the meantime, I think I prefer this (passing in a meaningless value that is the right type) to the previous status quo (passing in a meaningless value that is also the wrong type.)